fix compile when using clang as $CC:

Based on patch from https://bugs.launchpad.net/hugin/+bug/1213585 (rebased)

diff -r 2a43e83684d5 src/foreign/zthread/include/zthread/Guard.h
--- a/include/zthread/Guard.h	Sat Aug 10 11:31:46 2013 +0200
+++ b/include/zthread/Guard.h	Sun Aug 18 09:46:43 2013 +0200
@@ -108,7 +108,7 @@
   }
 
   template <class LockType>
-  static void createScope(LockHolder<LockType>& l, unsigned long ms) {
+  static bool createScope(LockHolder<LockType>& l, unsigned long ms) {
 
     if(Scope1::createScope(l, ms))
       if(!Scope2::createScope(l, ms)) {
